spec/services/bucket_service_spec.rb

```diff
@@ -5,17 +5,106 @@
+
+      # create member without balance, who has subscribed to participant activity + has commented on bucket
+      @subscribed_participant_without_balance = create_bucket_participant(bucket: bucket, subscribed: true)
+
+      # bucket creator comments on their bucket
+      create(:comment, bucket: bucket, user: user)
+
+      # create bucket participant not subscribed to participant activity
+      @unsubscribed_participant = create_bucket_participant(bucket: bucket, subscribed: false)
+
+      # create non participant
+      @subscribed_non_participant = create(:user, subscribed_to_participant_activity: true)
+      create(:membership, group: group, member: @subscribed_non_participant)
+
+      # create an archived participant
+      @archived_participant = create_bucket_participant(bucket: bucket, subscribed: true)
+      Membership.find_by(group: bucket.group, member: @archived_participant).update(archived_at: DateTime.now.utc - 5.days)
+
+      BucketService.send_bucket_live_emails(bucket: bucket)
+      @emails = ActionMailer::Base.deliveries
+      @email_recipients = @emails.map { |e| e.to.first }
+    end
+
+    it "sends emails to all bucket participants subscribed to participant activity, whose membership in the group is active, except bucket author" do
+      expect(@emails.length).to eq(2)
+      expect(@email_recipients).to include(@subscribed_participant_with_balance.email)
+      expect(@email_recipients).to include(@subscribed_participant_without_balance.email)      
+
+      expect(@email_recipients).not_to include(@bucket_author.email)
+      expect(@email_recipients).not_to include(@unsubscribed_participant.email)
+      expect(@email_recipients).not_to include(@subscribed_non_participant.email)
+      expect(@email_recipients).not_to include(@archived_participant)
+    end
+
+    it "participants with funds in the group get an email with their balance" do
+      email = @emails.find { |e| e.to.first ==  @subscribed_participant_with_balance.email }
+      expect(email.body).to include("$100")
+    end
+
+    it "participants without funds in the group get an email that does not contain their balance" do
+      email = @emails.find { |e| e.to.first ==  @subscribed_participant_without_balance.email }
+      expect(email.body).not_to include("$0")
+    end
+  end
+
+  describe "#send_bucket_funded_emails" do
+    before do
+      make_user_group_member
+      @bucket = create(:funded_bucket, user: user, group: group)
+      @bucket_author = @bucket.user
+
+      # create member with $100.00, who has subscribed to participant activity + has commented on bucket
+      @subscribed_participant = create_bucket_participant(bucket: @bucket, subscribed: true)
+
+      # bucket creator comments on their bucket
+      create(:comment, bucket: @bucket, user: @bucket_author)
+
+      # create bucket participant not subscribed to participant activity
```

Comment:
I think these instances of "not subscribed"   should be replaced with "unsubscribed"   .. i got a little confused to start with. 
not big deal though